const declarations should be order independent

**V version:** V 0.1.17
**OS:** manjaro linux

**What did you do?**
```go
const (
	a = b
	b = 1
)

fn main() {
	println(a)
}
```

**What did you expect to see?**
I expected both a and b to be a constant value of 1.
 
**What did you see instead?**
```
pass=2 fn=`main`
panic: test.v:7
Fn "println" wrong arg #1. Expected "string" (s)  but got "void"
```
Note: swapping the order of ```a = b``` and ```b = 1``` works as expected (both are 1). I suggest that constants should be order independent similar to functions.

--- a/vlib/compiler/parser.v
+++ b/vlib/compiler/parser.v
@@ -534,6 +534,21 @@ fn (p mut Parser) const_decl() {
 		if p.first_pass() {
 			p.table.register_const(name, typ, p.mod)
 		}
+		// Check to see if this constant exists, and is void. If so, try and get the type again:
+		for { // TODO: Find out how the error handling, and use it here instead of the for/break hack...
+			my_const := p.v.table.find_const(name) or {
+				break
+			} 
+			if my_const.typ == 'void' {
+				for i, v in p.v.table.consts {
+					if v.name == name {
+						p.v.table.consts[i].typ = typ
+						break
+					}
+				}
+			}
+			break
+		}
 		if p.pass == .main && !p.cgen.nogen {
 			// TODO hack
 			// cur_line has const's value right now. if it's just a number, then optimize generation:

--- a/vlib/compiler/tests/const_test.v
+++ b/vlib/compiler/tests/const_test.v
@@ -1,0 +1,10 @@
+const (
+	// c = a // TODO
+	a = b
+	b = 1
+)
+
+fn test_const() {
+	assert a == 1
+	// assert c == 1 // TODO: This will not build yet
+}

